File: bookshop-back-api/bookshop.webapi/Models/OrderFolder/OrderItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace bookshop.webapi.Models.OrderFolder
{
    public class OrderItem
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public Book Book { get; set; }
        public Order Order { get; set; }
        public int Quantity { get; set; }
    }
}

File: bookshop-front-angular/src/models/OrderItem.ts
import { Book } from "./Book";

export class OrderItem {
    public id: number = 0;
    public book: Book | null = null;
    public quantity: number = 0;

    constructor(object: any) {
        this.id = object.id;
        this.book = object.book;
        this.quantity = object.quantity;
    }
}